src/app/reporting/Engine.cpp

```diff
@@ -52,6 +59,58 @@ Status EventPathValid(DataModel::Provider * model, const ConcreteEventPath & eve
     return Status::Success;
 }
 
+/// Returns the status of ACL validation.
+///   if the status is set, the status is FINAL (i.e. permanent failure OR success due to path expansion logic.)
+///   if the status is not set, the processing can continue
+std::optional<CHIP_ERROR> ValidateReadACL(DataModel::Provider * dataModel, const Access::SubjectDescriptor & subjectDescriptor,
+                                          const ConcreteReadAttributePath & path)
+{
+
+    Access::RequestPath requestPath{ .cluster     = path.mClusterId,
+                                     .endpoint    = path.mEndpointId,
+                                     .requestType = Access::RequestType::kAttributeReadRequest,
+                                     .entityId    = path.mAttributeId };
+
+    std::optional<DataModel::AttributeInfo> info = dataModel->GetAttributeInfo(path);
+
+    chip::Access::Privilege requiredPrivilege = chip::Access::Privilege::kView; // default
+    if (info.has_value() && info->readPrivilege.has_value())
+    {
+        // set a default even if we do not know and later report the real error if ACL looks ok
+        requiredPrivilege = *info->readPrivilege;
+    }
+
+    CHIP_ERROR err = Access::GetAccessControl().Check(subjectDescriptor, requestPath, requiredPrivilege);
+    if (err == CHIP_NO_ERROR)
+    {
+        if (IsSupportedGlobalAttributeNotInMetadata(path.mAttributeId))
+        {
+            // Global attributes passing a kView check is ok
+            return std::nullopt;
+        }
+
+        // attribute does not exist, however we do not now if this is a unsupported endpoint, cluster or attribute,
```

Comment:
Why does attribute not exist?  all we know is we did an access check and it succeeeded.

This code is not making any sense to me.  Stopping here.